predictors/process.py

```diff
@@ -0,0 +1,42 @@
+import numpy as np
+import pandas as pd
+
+
+def add_all_predictors(predictor_subset, columns):
+    """Add in all predictors in columns to predictor_subset."""
+    other_predictors = []
+    for predictor in columns:
+        if predictor not in predictor_subset and 'FCH4' not in predictor:
+           other_predictors.append(predictor)
+        elif predictor == 'FCH4':
+            continue
+        elif 'FCH4' in predictor:
+            print(f"Ignoring predictor {predictor}.")
+
+    predictor_subset += other_predictors
+    predictor_subset.remove('all')
+    
+    return predictor_subset
+
+
+def process_wind_direction_predictor(X):
+    X = X.copy()
```

Comment:
Why did you do this?

Reply:
It's not crucial to do this, but I typically like to do this to avoid functions modifying variables in place. Specifically, a user might call the `process_wind_direction_predictor` on some dataframe `X`, and not expect the function to change the actual contents of the dataframe they passed in. The `.copy()` prohibits the function from changing the variable passed in as it makes a copy and operates on the copy instead of the original variable.